src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
here we can avoid the usage of ternary operator, since it makes it difficult to read the code.

here is a implementation of a alternative solution:

```js
let women = [];

if (withChildren) {
  women = people.filter((p) => p.sex === 'f'
    && people.find((p1) => p1.mother === p.name));
} else {
  women = people.filter((p) => p.sex === 'f');
}
```

to avoid nested method usage, we can split the operation into separate steps. This makes the code more readable and easier to maintain.

```js
const mothers = people.map((p) => p.mother).filter(Boolean);

const women = people.filter((p) => {
  const isFemale = p.sex === 'f';
  const isMother = mothers.includes(p.name);
  return isFemale && isMother;
});
```